### PR TITLE
Multiple fix of the maven build

### PR DESCRIPTION
**Pull request by henri-tremblay**
I mainly synched the pom with the offheap-store. Then javadoc errors were preventing compilation on Java 8. And a little library update to be a nice boyscout

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -63,9 +62,15 @@
       <scope>test</scope>
     </dependency>
     <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-jdk14</artifactId>
```
Comment (ljacomet) [inline pom.xml]: why move it around?

File: src/main/java/org/terracotta/statistics/jsr166e/LongAdder.java [hunk 2]
```diff
@@ -33,7 +33,7 @@
  * collection keys.
  *
  * <p><em>jsr166e note: This class is targeted to be placed in
- * java.util.concurrent.atomic<em>
+ * java.util.concurrent.atomic</em>
```
Comment (chrisdennis) [inline src/main/java/org/terracotta/statistics/jsr166e/LongAdder.java]: This is a lift from the jsr166 CVS repo. I'd like to keep it unmodified.

File: src/main/java/org/terracotta/statistics/jsr166e/LongMaxUpdater.java [hunk 3]
```diff
@@ -24,7 +24,7 @@
  * collection keys.
  *
  * <p><em>jsr166e note: This class is targeted to be placed in
- * java.util.concurrent.atomic<em>
+ * java.util.concurrent.atomic</em>
```
Comment (chrisdennis) [inline src/main/java/org/terracotta/statistics/jsr166e/LongMaxUpdater.java]: Ditto... not our file.

File: src/main/java/org/terracotta/statistics/jsr166e/ThreadLocalRandom.java [hunk 4]
```diff
@@ -23,8 +23,7 @@
  * than shared {@code Random} objects in concurrent programs will
  * typically encounter much less overhead and contention.  Use of
  * {@code ThreadLocalRandom} is particularly appropriate when multiple
- * tasks (for example, each a {@link ForkJoinTask}) use random numbers
- * in parallel in thread pools.
+ * tasks use random numbers in parallel in thread pools.
```
Comment (chrisdennis) [inline src/main/java/org/terracotta/statistics/jsr166e/ThreadLocalRandom.java]: Not ours.

File: src/main/java/org/terracotta/statistics/util/VicariousThreadLocal.java [hunk 5]
```diff
@@ -23,7 +23,7 @@
  * A drop-in replacement {@code ThreadLocal} implementation that does not leak
  * when thread-local values reference the {@code ThreadLocal} object.
  * The code is optimised to cope with frequently changing values.
- * <p/>
+ * <p>
```
Comment (chrisdennis) [inline src/main/java/org/terracotta/statistics/util/VicariousThreadLocal.java]: Not ours.

File: src/main/java/org/terracotta/statistics/jsr166e/LongAdder.java (hunk 2 above)
Comment (ljacomet) [inline src/main/java/org/terracotta/statistics/jsr166e/LongAdder.java]: Not too fond of modifying these classes which have been ported from another code base. Especially to please the Java 8 javadoc.

File: pom.xml [hunk 6]
```diff
@@ -22,38 +22,37 @@
   <description> A statistics framework used inside Ehcache and the Terracotta products</description>
   <url>https://github.com/Terracotta-OSS/terracotta-management/</url>
 
+  <prerequisites>
```
Comment (ljacomet) [inline pom.xml]: Since you go around fixing the build - please use the toolchain system to force it to be built with Java 6.

File: src/main/java/org/terracotta/statistics/jsr166e/LongAdder.java (hunk 2 above)
Comment (henri-tremblay) [inline src/main/java/org/terracotta/statistics/jsr166e/LongAdder.java]: Yes. Me too. The problem is that Doug had errors in his javadoc. So it doesn't compile. Let me try just to exclude the package. Might be wiser.

File: pom.xml (hunk 6 above)
Comment (henri-tremblay) [inline pom.xml]: I don't think other terracotta-oss are using the toolchain no? offheap-store doesn't. And that can be another PR :-)

**Comment by henri-tremblay** [comment]
Ok. I now ignore jsr166 package. Better indeed

File: src/main/java/org/terracotta/statistics/util/VicariousThreadLocal.java (hunk 5 above)
Comment (henri-tremblay) [inline src/main/java/org/terracotta/statistics/util/VicariousThreadLocal.java]: This is in the `util` package. So it seems to be ours.

Comment (chrisdennis) [inline src/main/java/org/terracotta/statistics/util/VicariousThreadLocal.java]: In one of our packages, but not our code.
```
/**
 * Copyright 2006 Thomas Hawtin
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
```
